client/src/modules/profiles/pages/userProfile.tsx

```diff
@@ -94,9 +110,57 @@ export const UserProfilePage = () => {
             loadingText={'Saving Profile Changes'}
             submitText={'Save Profile Changes'}
           />
-          <Button colorScheme={'red'} marginBlock={'2em'} onClick={clickDelete}>
-            Delete My Data
-          </Button>
+          <Flex gap="2em" marginBlock={'2em'}>
+            <Button
+              height={'2.8em'}
+              paddingBlock={'.65em'}
+              paddingInline={'.4em'}
+              colorScheme={'red'}
+              onClick={clickDelete}
+            >
+              Delete My Data
+            </Button>
+            <Button
+              fontWeight="600"
+              background={'gray.85'}
+              color={'gray.10'}
+              height={'100%'}
+              size={'lg'}
+              borderRadius={'5px'}
+              paddingBlock={'.65em'}
+              paddingInline={'.4em'}
+              _hover={{ color: 'gray.85', backgroundColor: 'gray.10' }}
+              onClick={() => getData()}
+              isDisabled={!!userData}
+            >
+              Request your data
+            </Button>
+            {loading ? (
+              <Spinner />
+            ) : (
+              <>
+                {userData && (
+                  <Flex alignItems={'center'} gap="2em">
+                    <Text>The data was fetched succesfully 🥳</Text>
+                    <Link
+                      fontWeight="600"
+                      background={'gray.85'}
+                      color={'gray.10'}
+                      height={'100%'}
+                      size={'lg'}
+                      borderRadius={'5px'}
+                      paddingBlock={'.65em'}
+                      paddingInline={'.4em'}
+                      download={`${userData?.name}.json`}
+                      href={createDownloadData(userData)}
```

Comment:
I have no idea about the logic behind this error in this line, I have checked the owner.json download file and this is the result

```json
{
    "id": 1,
    "name": "The Owner",
    "email": "foo@bar.com",
    "auto_subscribe": false,
    "image_url": null,
    "instance_role": {
        "name": "owner"
    },
    "user_bans": [],
    "user_chapters": [
        {
            "subscribed": true,
            "chapter_role": {
                "name": "member"
            },
            "chapter": {
                "id": 1,
                "name": "Heathcote, Koch and Upton"
            }
        },
        {
            "subscribed": true,
            "chapter_role": {
                "name": "member"
            },
            "chapter": {
                "id": 2,
                "name": "Rohan, Franey and Cole"
            }
        },
        {
            "subscribed": true,
            "chapter_role": {
                "name": "member"
            },
            "chapter": {
                "id": 3,
                "name": "Cremin Group"
            }
        },
        {
            "subscribed": true,
            "chapter_role": {
                "name": "member"
            },
            "chapter": {
                "id": 4,
                "name": "Lynch - Bartoletti"
            }
        }
    ],
    "user_events": []
}
```

I am getting this error.

> Argument of type '{ __typename?: \"UserData\" | undefined; id: number; name: string; email: string; auto_subscribe: boolean; image_url?: string | null | undefined; instance_role: { __typename?: \"InstanceRole\" | undefined; name: string; }; user_bans: { ...; }[]; user_chapters: { ...; }[]; user_events: { ...; }[]; }' is not assignable to parameter of type 'UserProfileQuery'.\n  Types of property '__typename' are incompatible.\n    Type '\"UserData\" | undefined' is not assignable to type '\"Query\" | undefined'.\n      Type '\"UserData\"' is not assignable to type '\"Query\"'.

Reply:
```ts
{
    "id": 1,
    "name": "The Owner",
    "email": "foo@bar.com",
    "auto_subscribe": false,
    "image_url": null,
    "admined_chapters": [
        {
            "id": 1,
            "name": "Bradtke - D'Amore",
            "category": "deserunt",
            "city": "Smithamton",
            "country": "Algeria",
            "creator_id": 1,
            "description": "beatae non nostrum",
            "region": "Vermont"
        },
        {
            "id": 2,
            "name": "Orn and Sons",
            "category": "facere",
            "city": "Deckowfield",
            "country": "Martinique",
            "creator_id": 1,
            "description": "neque enim consequatur",
            "region": "Kentucky"
        },
        {
            "id": 3,
            "name": "Rau, Metz and Kulas",
            "category": "aspernatur",
            "city": "Lake Percyton",
            "country": "Cook Islands",
            "creator_id": 1,
            "description": "voluptatem eveniet mollitia",
            "region": "Pennsylvania"
        },
        {
            "id": 4,
            "name": "Kreiger LLC",
            "category": "ut",
            "city": "Port Kennedi",
            "country": "Libyan Arab Jamahiriya",
            "creator_id": 1,
            "description": "a voluptate nisi",
            "region": "Minnesota"
        }
    ],
    "instance_role": {
        "id": 3,
        "name": "owner",
        "instance_role_permissions": [
            {
                "instance_permission": {
                    "name": "chapter-create"
                }
            },
            {
                "instance_permission": {
                    "name": "chapter-join"
                }
            },
            {
                "instance_permission": {
                    "name": "chapter-delete"
                }
            },
            {
                "instance_permission": {
                    "name": "chapter-subscription-manage"
                }
            },
            {
                "instance_permission": {
                    "name": "chapter-user-role-change"
                }
            },
            {
                "instance_permission": {
                    "name": "sponsor-manage"
                }
            },
            {
                "instance_permission": {
                    "name": "sponsor-view"
                }
            },
            {
                "instance_permission": {
                    "name": "user-instance-role-change"
                }
            },
            {
                "instance_permission": {
                    "name": "users-view"
                }
            },
            {
                "instance_permission": {
                    "name": "google-authenticate"
                }
            },
            {
                "instance_permission": {
                    "name": "chapter-edit"
                }
            },
            {
                "instance_permission": {
                    "name": "chapter-ban-user"
                }
            },
            {
                "instance_permission": {
                    "name": "event-create"
                }
            },
            {
                "instance_permission": {
                    "name": "event-edit"
                }
            },
            {
                "instance_permission": {
                    "name": "event-delete"
                }
            },
            {
                "instance_permission": {
                    "name": "event-send-invite"
                }
            },
            {
                "instance_permission": {
                    "name": "event-subscription-manage"
                }
            },
            {
                "instance_permission": {
                    "name": "rsvp"
                }
            },
            {
                "instance_permission": {
                    "name": "rsvp-delete"
                }
            },
            {
                "instance_permission": {
                    "name": "rsvp-confirm"
                }
            },
            {
                "instance_permission": {
                    "name": "venue-create"
                }
            },
            {
                "instance_permission": {
                    "name": "venue-edit"
                }
            },
            {
                "instance_permission": {
                    "name": "venue-delete"
                }
            },
            {
                "instance_permission": {
                    "name": "chapters-view"
                }
            },
            {
                "instance_permission": {
                    "name": "events-view"
                }
            },
            {
                "instance_permission": {
                    "name": "venues-view"
                }
            }
        ]
    },
    "user_bans": [],
    "user_chapters": [
        {
            "subscribed": true,
            "chapter_role": {
                "id": 1,
                "name": "member",
                "chapter_role_permissions": [
                    {
                        "chapter_permission": {
                            "id": 7,
                            "name": "event-subscription-manage"
                        }
                    },
                    {
                        "chapter_permission": {
                            "id": 8,
                            "name": "rsvp"
                        }
                    }
                ]
            },
            "chapter": {
                "id": 1,
                "name": "Bradtke - D'Amore",
                "description": "beatae non nostrum",
                "category": "deserunt",
                "city": "Smithamton",
                "region": "Vermont",
                "country": "Algeria",
                "creator_id": 1
            },
            "chapter_id": 1,
            "joined_date": "2022-12-01T12:08:51.324Z",
            "user": {
                "id": 1,
                "name": "The Owner",
                "auto_subscribe": false
            },
            "user_id": 1
        },
        {
            "subscribed": true,
            "chapter_role": {
                "id": 1,
                "name": "member",
                "chapter_role_permissions": [
                    {
                        "chapter_permission": {
                            "id": 7,
                            "name": "event-subscription-manage"
                        }
                    },
                    {
                        "chapter_permission": {
                            "id": 8,
                            "name": "rsvp"
                        }
                    }
                ]
            },
            "chapter": {
                "id": 2,
                "name": "Orn and Sons",
                "description": "neque enim consequatur",
                "category": "facere",
                "city": "Deckowfield",
                "region": "Kentucky",
                "country": "Martinique",
                "creator_id": 1
            },
            "chapter_id": 2,
            "joined_date": "2022-12-01T12:08:51.326Z",
            "user": {
                "id": 1,
                "name": "The Owner",
                "auto_subscribe": false
            },
            "user_id": 1
        },
        {
            "subscribed": true,
            "chapter_role": {
                "id": 1,
                "name": "member",
                "chapter_role_permissions": [
                    {
                        "chapter_permission": {
                            "id": 7,
                            "name": "event-subscription-manage"
                        }
                    },
                    {
                        "chapter_permission": {
                            "id": 8,
                            "name": "rsvp"
                        }
                    }
                ]
            },
            "chapter": {
                "id": 3,
                "name": "Rau, Metz and Kulas",
                "description": "voluptatem eveniet mollitia",
                "category": "aspernatur",
                "city": "Lake Percyton",
                "region": "Pennsylvania",
                "country": "Cook Islands",
                "creator_id": 1
            },
            "chapter_id": 3,
            "joined_date": "2022-12-01T12:08:51.327Z",
            "user": {
                "id": 1,
                "name": "The Owner",
                "auto_subscribe": false
            },
            "user_id": 1
        },
        {
            "subscribed": true,
            "chapter_role": {
                "id": 1,
                "name": "member",
                "chapter_role_permissions": [
                    {
                        "chapter_permission": {
                            "id": 7,
                            "name": "event-subscription-manage"
                        }
                    },
                    {
                        "chapter_permission": {
                            "id": 8,
                            "name": "rsvp"
                        }
                    }
                ]
            },
            "chapter": {
                "id": 4,
                "name": "Kreiger LLC",
                "description": "a voluptate nisi",
                "category": "ut",
                "city": "Port Kennedi",
                "region": "Minnesota",
                "country": "Libyan Arab Jamahiriya",
                "creator_id": 1
            },
            "chapter_id": 4,
            "joined_date": "2022-12-01T12:08:51.328Z",
            "user": {
                "id": 1,
                "name": "The Owner",
                "auto_subscribe": false
            },
            "user_id": 1
        }
    ],
    "user_events": []
}
```